pkg/prober/prober_windows.go

```diff
@@ -43,9 +43,11 @@ func GetSystemCertPool(probeName string) (*x509.CertPool, error) {
 	var certContext *syscall.CertContext
 
 	// ref for why flags value is 0: https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certclosestore
-	defer func(store syscall.Handle, flags uint32) {
-		_ = syscall.CertCloseStore(store, flags)
-	}(certContext.Store, 0)
+	defer func() {
```

Comment:
The defer needs to be moved below the for loop on L94. I missed this on my initial PR, sorry!

Reply:
Why does it need to be moved?

Reply:
in addition to moving the defer, you would also do your original nil check after the for loop